File: typescript/payments/raw/sandbox-advance.ts
import { PagouHttpClient } from "../../src/lib/http.js";
import { printResult } from "../../src/lib/format.js";
import type { TransactionStatus } from "../../src/lib/types.js";
import { resourceIdFromArgs } from "../demo-data.js";

// Sandbox-only helper: forces a transaction to a target status so you can
// exercise the paid/refunded paths without a real payer. Never available in
// production. Run: npm run pay:pix, then:
// npx tsx payments/raw/sandbox-advance.ts <transaction_id> [status=paid]
async function main(): Promise<void> {
  const id = resourceIdFromArgs("PAGOU_TRANSACTION_ID");
  const status = (process.argv[3] ?? "paid") as TransactionStatus;
  const client = new PagouHttpClient();

  const { data } = await client.requestData<{ transaction: { id: string; status: string } }>({
    method: "PUT",
    path: `/v2/transactions/${id}`,
    body: { status },
  });

  printResult("Sandbox transaction updated", data.transaction);
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: java/src/main/java/ai/pagou/examples/payments/SandboxAdvance.java
package ai.pagou.examples.payments;

import ai.pagou.examples.lib.Config;
import ai.pagou.examples.lib.Format;
import ai.pagou.examples.lib.PagouHttpClient;
import ai.pagou.examples.lib.Request;
import com.fasterxml.jackson.databind.JsonNode;
import java.util.Map;

// Sandbox-only helper: forces a transaction to a target status so you can
// exercise the paid/refunded paths without a real payer. Never available in
// production.
// Run: mvn -q compile exec:java@pay-sandbox-advance -Dexec.args="<transaction_id> [status=paid]"
public final class SandboxAdvance {

  private SandboxAdvance() {}

  public static void main(String[] args) {
    String id = Config.resourceIdFromArgs(args, "PAGOU_TRANSACTION_ID");
    String status = args.length > 1 && !args[1].isBlank() ? args[1] : "paid";
    PagouHttpClient client = new PagouHttpClient();

    JsonNode data =
        client
            .requestData(
                Request.put("/v2/transactions/" + id).body(Map.of("status", status)), JsonNode.class)
            .data();

    Format.printResult("Sandbox transaction updated", data.has("transaction") ? data.get("transaction") : data);
  }
}
